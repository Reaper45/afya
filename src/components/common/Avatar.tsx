import * as React from "react";
import { ImgHTMLAttributes } from "react";

import styled from "lib/emotion";

const AvatarWrapper = styled("div")`
  display: flex;
  align-items: center;
`;

const TextWrapper = styled("div")`
  margin-right: 1em;
  flex-grow: 1;
`;

const ImageWrapper = styled("img")`
  border-radius: 1em;
  width: 40px;
  height: 40px;
`;

const AvatarTitle = styled("b")`
  display: block;
  font-size: 0.8em;
  color: #222222;
`;

const Description = styled("small")`
  color: ${props => props.theme.colors.accent};
  display: block;
  margin-top: 0.3rem;
  font-size: 0.7em;
`;

const Avatar: React.FC<ImgHTMLAttributes<HTMLImageElement> & {
  description?: string;
  title?: string;
  img: string;
}> = ({ description, title, img }) => {
  return (
    <AvatarWrapper>
      {(description || title) && (
        <TextWrapper>
          <AvatarTitle data-name="title">{title}</AvatarTitle>
          <Description data-name="description">{description}</Description>
        </TextWrapper>
      )}
      <div></div>
      <ImageWrapper src={img} alt="" />
    </AvatarWrapper>
  );
};

export default Avatar;
