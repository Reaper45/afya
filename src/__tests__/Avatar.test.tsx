import * as React from "react";
import { shallow } from "enzyme";

import Avatar from "components/common/Avatar";

import avatar from "static/avatar.jpg";

it("renders correctly", () => {
  const component = shallow(<Avatar img={avatar} />);
  expect(component).toMatchSnapshot();
});

it("should renders title", () => {
  const component = shallow(<Avatar img={avatar} title="Sandra Mamai" />);
  expect(component.find('[data-name="title"]').text()).toEqual("Sandra Mamai");
});

it("should renders description", () => {
  const component = shallow(
    <Avatar img={avatar} description="sandra.mamai@test.com" />
  );
  expect(component.find('[data-name="description"]').text()).toEqual(
    "sandra.mamai@test.com"
  );
});

