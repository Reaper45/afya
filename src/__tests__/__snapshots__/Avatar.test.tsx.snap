// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Styled(div)>
  <div />
  <Styled(img)
    alt=""
    src="avatar.jpg"
  />
</Styled(div)>
`;
